Mathlib/CategoryTheory/Sites/EffectiveEpimorphic.lean

```diff
@@ -454,42 +443,44 @@ theorem Sieve.effectiveEpimorphic_family {B : C} {α : Type*}
 
 section instances
 
+/--
+Given an `EffectiveEpiFamily X π` and a corresponding coproduct cocone, the family descends to an
+`EffectiveEpi` from the coproduct.
+-/
+noncomputable
+def effectiveEpiStructIsColimitDescOfEffectiveEpiFamily {B : C} {α : Type*} (X : α → C)
+    (c : Cofan X) (hc : IsColimit c) (π : (a : α) → (X a ⟶ B)) [EffectiveEpiFamily X π] :
+    EffectiveEpiStruct (hc.desc (Cofan.mk B π)) where
+  desc e h := EffectiveEpiFamily.desc X π (fun a ↦ c.ι.app ⟨a⟩ ≫ e) (fun a₁ a₂ g₁ g₂ hg ↦ by
+    simp only [← Category.assoc]
+    exact h (g₁ ≫ c.ι.app ⟨a₁⟩) (g₂ ≫ c.ι.app ⟨a₂⟩) (by simpa))
+  fac e h := hc.hom_ext (fun ⟨j⟩ ↦ (by simp))
+  uniq e _ m hm := EffectiveEpiFamily.uniq X π (fun a ↦ c.ι.app ⟨a⟩ ≫ e)
+      (fun _ _ _ _ hg ↦ (by simp [← hm, reassoc_of% hg])) m (fun _ ↦ (by simp [← hm]))
+
 /--
 Given an `EffectiveEpiFamily X π` such that the coproduct of `X` exists, `Sigma.desc π` is an
 `EffectiveEpi`.
 -/
 noncomputable
 def effectiveEpiStructDescOfEffectiveEpiFamily {B : C} {α : Type*} (X : α → C)
     (π : (a : α) → (X a ⟶ B)) [HasCoproduct X] [EffectiveEpiFamily X π] :
-    EffectiveEpiStruct (Sigma.desc π) where
-  desc e h := EffectiveEpiFamily.desc X π (fun a ↦ Sigma.ι X a ≫ e) (fun a₁ a₂ g₁ g₂ hg ↦ by
-    simp only [← Category.assoc]
-    apply h (g₁ ≫ Sigma.ι X a₁) (g₂ ≫ Sigma.ι X a₂)
-    simpa only [Category.assoc, colimit.ι_desc, Cofan.mk_pt, Cofan.mk_ι_app])
-  fac e h := by
-    ext a
-    simp only [colimit.ι_desc_assoc, Discrete.functor_obj, Cofan.mk_pt, Cofan.mk_ι_app,
-      EffectiveEpiFamily.fac]
-  uniq e _ m hm := by
-    have := EffectiveEpiFamily.uniq X π (fun a ↦ Sigma.ι X a ≫ e) ?_ m
-    · apply this
-      intro a
-      rw [← hm]
-      simp only [colimit.ι_desc_assoc, Discrete.functor_obj, Cofan.mk_pt, Cofan.mk_ι_app]
-    · intro Z a₁ a₂ g₁ g₂ hg
-      rw [← hm]
-      simp only [colimit.ι_desc_assoc, Discrete.functor_obj, Cofan.mk_pt, Cofan.mk_ι_app]
-      rw [← Category.assoc, hg, Category.assoc]
+    EffectiveEpiStruct (Sigma.desc π) := by
+  simpa [coproductIsCoproduct] using
+    effectiveEpiStructIsColimitDescOfEffectiveEpiFamily X _ (coproductIsCoproduct _) π
 
 instance {B : C} {α : Type*} (X : α → C) (π : (a : α) → (X a ⟶ B)) [HasCoproduct X]
     [EffectiveEpiFamily X π] : EffectiveEpi (Sigma.desc π) :=
   ⟨⟨effectiveEpiStructDescOfEffectiveEpiFamily X π⟩⟩
 
+-- This is to show that the instance `epiCoproductDescOfEffectiveEpiFamily` is not needed anymore.
```

Comment:
As `epiCoproductDescOfEffectiveEpiFamily` is no longer in the code, you may remove this line:
```suggestion
```